sheerwater_benchmarking/data/tahmo.py

```diff
@@ -0,0 +1,183 @@
+    lats, lons, grid_size = get_grid(grid)
+    # This rounding only works for divisible, uniform grids
+    assert (lats[0] % grid_size == 0)
+    assert (lons[0] % grid_size == 0)
+
+    def custom_round(x, base):
+        return base * round(float(x)/base)
+
+    stat = dd.read_csv("gs://sheerwater-datalake/tahmo-data/tahmo_station_locs.csv")
+    stat = stat.rename(columns={'latitude': 'lat', 'longitude': 'lon', 'station code': 'station_code'})
+    stat['lat'] = stat['lat'].apply(lambda x: custom_round(x, base=grid_size))
+    stat['lon'] = stat['lon'].apply(lambda x: custom_round(x, base=grid_size))
+
+    if cell_aggregation == 'first':
+        stations_to_use = stat.groupby(['lat', 'lon']).agg(station_code=('station_code', 'first'))
+        stations_to_use = stations_to_use.reset_index()
+        stations_to_use = stations_to_use['station_code'].unique()
+
+        stat = stat[stat['station_code'].isin(stations_to_use)]
+        stat = stat.set_index('station_code')
+        obs = obs.join(stat, on='station_code', how='inner')
+
+        # Prepare for xarray
+        obs = obs.drop(['station_code', 'name', 'country'], axis=1)
+
+        # Group by only way to set a multi index in dask?
+        obs = obs.groupby(['lat', 'lon', 'time']).agg(precip=('precip', 'mean'))
+    elif cell_aggregation == 'mean':
+        # Group by lat/lon/time
+        stat = stat.set_index('station_code')
+        obs = obs.join(stat, on='station_code', how='inner')
+        obs = obs.groupby(by=['lat', 'lon', 'time']).agg(precip=('precip', 'mean'))
+
+    # Convert to xarray - for this to succeed obs must be a pandas dataframe
+    obs = xr.Dataset.from_dataframe(obs.compute())
+
+    obs = obs.chunk({'time': 365,
+                     'lat': 300,
+                     'lon': 300})
+
+    # Return the xarray
+    return obs
+
+
+@dask_remote
+@cacheable(data_type='array',
+           timeseries='time',
+           cache_args=['grid', 'agg_days', 'missing_thresh', 'cell_aggregation'],
+           chunking={'lat': 300, 'lon': 300, 'time': 365})
+def tahmo_rolled(start_time, end_time, agg_days, grid='global0_25', missing_thresh=0.5, cell_aggregation='first'):
+    """Tahmo rolled and aggregated."""
+    # Get the data
+    ds = tahmo_raw(start_time, end_time, grid, cell_aggregation)
+
+    # Reindex to fill out the lat/lon
+    grid_ds = get_grid_ds(grid)
+    ds = ds.reindex_like(grid_ds)
+
+    # Roll and agg
+    agg_thresh = max(int(agg_days*missing_thresh), 1)
+
+    ds = roll_and_agg(ds, agg=agg_days, agg_col="time", agg_fn='mean', agg_thresh=agg_thresh)
+
+    return ds
+
+
+@dask_remote
+@cacheable(data_type='array',
+           timeseries='time',
+           cache_args=['variable', 'agg_days', 'grid', 'mask', 'region', 'missing_thresh'],
+           chunking={'lat': 300, 'lon': 300, 'time': 365},
+           cache=False)
+def tahmo(start_time, end_time, variable, agg_days, grid='global0_25', mask='lsm', region='global',
+         missing_thresh=0.5):
```

Comment:
For these accessors, I think having the standard format without other params is nice - I'd still vote to remove missing_thresh from this, and just set it to 0.5 in the code below .